ElectronClient/keymapHandler.ts

```diff
@@ -0,0 +1,130 @@
+const { shim } = require('lib/shim.js');
+
+const modifiers = /^(Command|Cmd|Control|Ctrl|CommandOrControl|CmdOrCtrl|Alt|Option|AltGr|Shift|Super)$/;
+const keyCodes = /^([0-9A-Z)!@#$%^&*(:+<_>?~{|}";=,\-./`[\\\]']|F1*[1-9]|F10|F2[0-4]|Plus|Space|Tab|Backspace|Delete|Insert|Return|Enter|Up|Down|Left|Right|Home|End|PageUp|PageDown|Escape|Esc|VolumeUp|VolumeDown|VolumeMute|MediaNextTrack|MediaPreviousTrack|MediaStop|MediaPlayPause|PrintScreen)$/;
+
+interface customKeymapItem {
+	accelerator: string;
+	command: string;
+}
+
+interface keymapItem extends customKeymapItem {
+	label: string;
+	section: string;
+	platforms?: string[];
+}
+
+const defaultKeymap: keymapItem[] = [
+	{ accelerator: 'CommandOrControl+N', command: 'newNoteItem', label: 'New note', section: 'File' },
```

Comment:
As this is a draft, I guess you plan to remove the label from here at some point? As discussed it shouldn't be here as a shortcut essentially maps some keys to an action. The *action* might have a label, which will be displayed in the menu, toolbar tooltip, etc. but that property shouldn't be defined here. Currently it's set mostly on the menu items so there's probably some refactoring to do so that this label and other properties of the action can be accessed from elsewhere.

Probably we should have structures like this:

```
const shortcut = { accelerator: "CommandOrControl+N", command: "newNoteItem", section: "File" }
```

and elsewhere:

```
const command = { name: "newNoteItem", label: _("New note") }
```

Reply:
I kept the labels so that the GUI editor can iterate over all shortcuts and display them. I guess we can remove the labels, and then hard-code command names and labels as you've mentioned in the GUI editor until we get the actions refactored?

Thanks for reviewing!

Reply:
I've removed `label` and `section` properties. I totally agree with your point here. This service should map an Accelerator to a command.

However, we need still to display labels for each command and group them by section in the GUI editor. I'm thinking maybe hard-coding them in the editor's code itself?

For example,
```{ command: 'newNoteItem', label: _("New note"), section: 'File' }```

I see no other option, at least until we get the actions centralized and sorted out. Would that approach make sense?